libraries/primitive_consul_health.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'primitive'
+require 'chef/mash'
+
+module Choregraphie
+  class ConsulHealthCheck < Primitive
+    def initialize(options = {})
+      @options = Mash.new(options)
+
+      @options[:tries] ||= 10
+      @options[:checkids] ||= []
+      @options[:delay] ||= 15
+
+      if @options[:consul_token]
+        require 'diplomat'
+        Diplomat.configure do |config|
+          config.acl_token = @options[:consul_token]
+        end
+      end
+
+      # this block could be used to configure diplomat if needed
+      yield if block_given?
+    end
+
+    def register(choregraphie)
+      choregraphie.cleanup do
+        require 'diplomat'
+        tries = @options[:tries]
+        all_passing = false
+        failing_checks = []
+
+        while !all_passing && tries != 0
```

Comment:
Agreed. I am not still used to ruby idioms such as `something.times` it is way easy to understand phrased this way

patching